examples/shader/post_processing.rs

```diff
@@ -0,0 +1,338 @@
+//! Shows how to render a custom post processing effect, here a chromatic aberration.
+//! This example is useful to implement your own post-processing effect such as
+//! edge detection, blur, pixelization, vignette... and countless others.
+
+use bevy::{
+    core_pipeline::{
+        draw_3d_graph, node, AlphaMask3d, Opaque3d, RenderTargetClearColors, Transparent3d,
+    },
+    ecs::system::{lifetimeless::SRes, SystemParamItem},
+    prelude::*,
+    reflect::TypeUuid,
+    render::{
+        camera::{ActiveCamera, Camera, CameraTypePlugin, RenderTarget},
+        render_asset::{PrepareAssetError, RenderAsset, RenderAssets},
+        render_graph::{Node, NodeRunError, RenderGraph, RenderGraphContext, SlotValue},
+        render_phase::RenderPhase,
+        render_resource::{
+            BindGroup, BindGroupDescriptor, BindGroupEntry, BindGroupLayout,
+            BindGroupLayoutDescriptor, BindGroupLayoutEntry, BindingResource, BindingType,
+            Extent3d, SamplerBindingType, ShaderStages, TextureDescriptor, TextureDimension,
+            TextureFormat, TextureSampleType, TextureUsages, TextureViewDimension,
+        },
+        renderer::{RenderContext, RenderDevice},
+        view::RenderLayers,
+        RenderApp, RenderStage,
+    },
+    sprite::{Material2d, Material2dPipeline, Material2dPlugin, MaterialMesh2dBundle},
+};
+
+#[derive(Component, Default)]
+pub struct FirstPassCamera;
+
+/// The name of the final node of the first pass.
+pub const FIRST_PASS_DRIVER: &str = "first_pass_driver";
```

Comment:
This pass is rather unnatural, this example would benefit having a new post-processing pass rather than a new "first pass" which replaces the main pass.

Reply:
do I did that, I think it's for the best.